core/src/stores/sqlite.rs

```diff
@@ -927,15 +927,21 @@ impl Store for SQLiteStore {
             let created = generation.created;
             let request_data = serde_json::to_string(&request)?;
             let generation_data = serde_json::to_string(&generation)?;
-            let _ = stmt.insert(params![
+            let statement = stmt.insert(params![
                 project_id,
                 created,
                 request.hash().to_string(),
                 request_data,
                 generation_data
-            ])?;
+            ]);
 
-            Ok(())
+            match statement {
+                Ok(_) => Ok(()),
+                Err(e) => match e.to_string().as_str() {
```

Comment:
I thiiiiink there is an error type for that, matching on that would be nice?

Reply:
Yeah it's better but didn't find the corresponding error here https://docs.rs/rusqlite/0.7.2/rusqlite/enum.Error.html

Reply:
moved the match upper 👍